src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -0,0 +1,197 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+
+namespace Internal.Runtime.InteropServices
+{
+    public static class ComponentActivator
+    {
+        public enum StringFormat
+        {
+            UTF8 = 0,
+            UTF16 = 1,
+            Ansi = 2
+        };
+
+        [StructLayout(LayoutKind.Sequential)]
+        [CLSCompliant(false)]
+        public unsafe struct CreateNativeDelegateArgs
```

Comment:
> adding unload support would be hard

It could be automatic.  We could only store the AssemblyLoadContext by weak reference, and rely on the delegate to keep it live.  When the delegate is released the AssemblyLoadContext could be collected.

Reply:
Unloading should be opt-in - it has a certain performance penalty on all code in the ALC when turned on.